packages/rest/src/lib/utils/types.ts

```diff
@@ -155,10 +155,20 @@ export interface RateLimitData {
 	 * The HTTP method being performed
 	 */
 	method: string;
+	/**
+	 * The time, in milliseconds, that will need to pass before this request can be retried
+	 */
+	retryAfter: number;
 	/**
 	 * The route being hit in this request
 	 */
 	route: string;
+	/**
+	 * The time, in milliseconds, that will need to pass before this request can be retried
```

Comment:
```suggestion
	 * The time, in milliseconds, that will need to pass before this specific request can be retried
```

Reply:
This is very much redundant to add 👀

Reply:
you need to somehow identify the difference between this, `retryAfter`, and `timeToReset`
`timeToReset` is pretty good already, but `retryAfter` and `sublimitTimeout` are not the same. A `sublimitTimeout` is guaranteed to be per-request, whereas `retryAfter` could be for the whole queue or per-request. In fact, I'd say we should change to `isSublimit: boolean`, since it'll either be the same value as `retryAfter` or not sublimited

Reply:
I just added all fields that also get logged in the debug message. In terms of pretty good already, for a 429 on the emojis route i mentioned, `timeToReset` was way wrong (always at around 120) while `retryAfter` and sublimit one were the correct, very long number.

Reply:
because as mentioned, the request-lock isn't tied to `retryAfter`. This is expected behavior; the number isn't wrong, it's the wrong number. On 429's, the current request is stalled, which will *technically* stall the whole queue. On encountering a sublimit, it *does not* stall the whole queue, because the next request is allowed to pass (if we can identify why we hit the sublimit at least)